pkg/pod/container.go

```diff
@@ -113,6 +113,81 @@ func (builder *ContainerBuilder) WithSecurityCapabilities(sCapabilities []string
 	return builder
 }
 
+// WithDropSecurityCapabilities drops SecurityCapabilities from the container definition.
+func (builder *ContainerBuilder) WithDropSecurityCapabilities(sCapabilities []string, redefine bool) *ContainerBuilder {
+	glog.V(100).Infof("Dropping a list of SecurityCapabilities %v from container %s",
+		sCapabilities, builder.definition.Name)
+
+	if !areCapabilitiesValid(sCapabilities) {
+		glog.V(100).Infof("Given SecurityCapabilities %v are not valid. Valid list %s",
+			sCapabilities, AllowedSCList)
+
+		builder.errorMsg = "one of the give securityCapabilities is invalid. Please extend allowed list or fix parameter"
+	}
+
+	var sCapabilitiesList []v1.Capability
+	for _, capability := range sCapabilities {
+		sCapabilitiesList = append(sCapabilitiesList, v1.Capability(capability))
+	}
+
+	// filter possible duplicated capabilities from user's input
+	sCapabilitiesList = uniqueCaps(sCapabilitiesList)
+	glog.V(100).Infof("Filtered user input: %v", sCapabilitiesList)
+
+	// filter conflicting caps between ADD and DROP
+	if builder.definition.SecurityContext != nil &&
+		builder.definition.SecurityContext.Capabilities != nil &&
+		builder.definition.SecurityContext.Capabilities.Add != nil {
+		glog.V(100).Infof("Filtering conflicting options between ADD and DROP capabilities")
+
+		confCapabilitiesList := capsDifference(
+			builder.definition.SecurityContext.Capabilities.Add,
+			sCapabilitiesList, false)
+
+		if len(confCapabilitiesList) > 0 {
+			glog.V(100).Infof("Conflicting ADD and DROP capabilities")
+
+			for _, mcap := range confCapabilitiesList {
+				glog.V(100).Infof("SecurityCapability %q already present in the Capabilities.Add list", mcap)
+			}
+
+			builder.errorMsg = "Conflicting ADD and DROP SecurityCapabilities"
+		}
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	if builder.definition.SecurityContext == nil {
+		glog.V(100).Infof("SecurityContext is nil initializing one")
+
+		builder.definition.SecurityContext = new(v1.SecurityContext)
+	}
+
+	if builder.definition.SecurityContext.Capabilities == nil {
+		glog.V(100).Infof("Capabilities are nil initializing one")
```

Comment:
These seem like 2 sentences (or am I wrong). Missing the period after nil.

Reply:
thanks, updated